bemani/backend/ddr/ddrx.py

```diff
@@ -0,0 +1,13 @@
+from typing import Optional
+
+from bemani.backend.ddr.base import DDRBase
+from bemani.backend.ddr.ddrsn2 import DDRSuperNova2
+from bemani.common import VersionConstants
+
+
+class DDRX(DDRBase):
```

Comment:
I had to extract this to fix a circular dependency? Maybe not needed?

Reply:
IDGI, where's the circular dependency? Does SN2 reference X for some reason? This isn't a problem anywhere else.

Reply:
I might try my hand at it again because I think it might have had something to do with how I was importing modules instead of a file directly? But it was mainly a problem with the `previous_version` function.

How the code used to be:
In Stub: `DDRX`, `DDRSN1`
Own Module: `DDRSN2`

`DDRX` has a `previous_version` that points to `DDRSN2` from the module, but `DDRSN2` has a `previous_version` that points to `DDRSN1` from the stub file. So the stub file needs the `DDRSN2` import to work, but that import can't work until the stub file resolves for `DDRSN1`? That's how I understood it but my imports might have been strange too, so let me review this one a bit more.

Reply:
Oh, it's because SN1 and X were in the stub file, so circular dependency. You moved the DDRX bits out of the stub into their own file, so it should work now.